pkg/controller/velerobackup/velerobackup_controller.go

```diff
@@ -151,5 +175,84 @@ func createVeleroBackupObject(client client.Client, namespace string) error {
 		},
 	}
 
-	return client.Create(context.TODO(), backup)
+	return r.Create(context.TODO(), backup)
+}
+
+func (r *ReconcileBackup) getModifiedHiveObjectsInNamespace(namespace string) ([]*hiveObject, error) {
+	nsLogger := addNamespaceLoggerFields(r.logger, namespace)
+
+	filteredHiveObjects := []*hiveObject{}
+
+	objList, err := getRuntimeObjects(r, typesToList, filterOnlyFromNamespace(namespace))
+	if err != nil {
+		nsLogger.WithError(err).Error("Failed to list hive objects in namespace.")
+		return nil, err
+	}
+
+	for _, nsObj := range objList {
+		hobj, err := newHiveObject(nsObj, r.logger)
+		if err != nil {
+			return nil, err
+		}
+
+		if hobj.hasChanged() {
+			filteredHiveObjects = append(filteredHiveObjects, hobj)
+		}
+	}
+
+	return filteredHiveObjects, nil
+}
+
+func (r *ReconcileBackup) updateHiveObjectsLastBackupChecksum(hobjs []*hiveObject) error {
+	for _, hobj := range hobjs {
+		hobjLogger := hobj.addLoggerFields()
+
+		hobj.setChecksumAnnotation()
+		obj := hobj.object
+
+		err := r.Update(context.TODO(), obj)
+		if err != nil {
+			hobjLogger.WithError(err).Error("Failed update.")
+			return err // If we error at all, then we should just return. The NS will be requeued in the future.
+		}
+	}
+
+	return nil
+}
+
+func addNamespaceLoggerFields(logger log.FieldLogger, namespace string) log.FieldLogger {
+	return logger.WithField("namespace", namespace)
+}
+
+type predicateType func(obj runtime.Object) bool
+
+func getRuntimeObjects(client client.Client, typesToList []runtime.Object, predicate predicateType) ([]runtime.Object, error) {
```

Comment:
Sorry, I did not think through my suggestion for this function thoroughly enough. This function should take a namespace as a parameter so that the call to `client.List` can do the namespace filtering. I was expecting the predicate to mimic what you had before where where there is one predicate that gets all of the changed object and one that gets all of the objects. But now that I think about it, that is not practical as it requires converting to a `hiveObject` prior to check whether the checksum has changed. So, the predicate is not necessary.

Reply:
Ah, that makes sense. Fixing.